code/datums/emotes.dm

```diff
@@ -0,0 +1,90 @@
+#define EMOTE_VISIBLE 1
+#define EMOTE_AUDIBLE 2
+
+var/global/list/emote_list = list()
+
+/datum/emote
+	var/key //What calls the emote
+	var/key_third_person //This will also call the emote
+	var/message //Message displayed when emote is used
+	var/message_mime //Message displayed if the user is a mime
+	var/message_alien //Message displayed if the user is a grown alien
+	var/message_larva //Message displayed if the user is an alien larva
+	var/message_robot //Message displayed if the user is a robot
+	var/message_param //Message to display if a param was given
+	var/emote_type = EMOTE_VISIBLE //Whether the emote is visible or audible
+	var/restraint_check = FALSE //Checks if the mob is restrained before performing the emote
+	var/muzzle_ignore = FALSE //Will only work if the emote is EMOTE_AUDIBLE
+	var/list/mob_type_allowed_typecache = list() //Types that are allowed to use that emote
+	var/list/mob_type_blacklist_typecache = list() //Types that are NOT allowed to use that emote
```

Comment:
This might sound super ultra pandemic, but could you initialized the ones that are supposed to be strings as ""? The idea is to init them as what they're supposed to be so a bool would be TRUE/FALSE, a number might be 0, strings are "", etc